phases/19-capstone-projects/37-loading-pretrained-weights/code/main.py

```
@@ -0,0 +1,396 @@
+        self.final_ln = LayerNorm(cfg.d_model)
+        self.lm_head = nn.Linear(cfg.d_model, cfg.vocab_size, bias=False)
+        if cfg.weight_tying:
+            self.lm_head.weight = self.tok_embed.weight
+        self.register_buffer(
+            "position_ids",
+            torch.arange(cfg.context_length, dtype=torch.long),
+            persistent=False,
+        )
+
+    def forward(self, tokens: torch.Tensor) -> torch.Tensor:
+        batch, seq = tokens.shape
```

Ruff (0.15.14)

[warning] Unpacked variable `batch` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)